src/sprites/AbilityPanel.lua

```diff
@@ -47,103 +47,56 @@ function AbilityPanel.getInstance() return _instance end
 
 function AbilityPanel:init()
   AbilityPanel.super.init(self, imagePanel)
-
   _instance = self
 
-  self:moveTo(0, panelShownY)
+  self:setCenter(0, 0)
+  self:moveTo(0, 0)
+  self:setZIndex(99)
   self:add()
-
-  spriteOne:moveTo(16, buttonShownY)
-  spriteOne:setZIndex(100)
-  spriteOne:add()
-  spriteTwo:moveTo(48, buttonShownY)
-  spriteTwo:setZIndex(100)
-  spriteTwo:add()
-  spriteThree:moveTo(80, buttonShownY)
-  spriteThree:setZIndex(100)
-  spriteThree:add()
-
   self:setIgnoresDrawOffset(true)
-  spriteOne:setIgnoresDrawOffset(true)
-  spriteTwo:setIgnoresDrawOffset(true)
-  spriteThree:setIgnoresDrawOffset(true)
-
-  self.abilitiesCount = 1
-  self:setZIndex(99)
-end
-
-local _spriteAdd = AbilityPanel.add
-function AbilityPanel:add()
-  _spriteAdd(self)
 
-  spriteOne:add()
-  spriteTwo:add()
-  spriteThree:add()
-end
-
-function AbilityPanel:shake(shakeTime, shakeMagnitude)
-  local shakeTimer = pd.timer.new(shakeTime, shakeMagnitude, 0)
-
-  shakeTimer.updateCallback = function(timer)
-    local magnitude = math.floor(timer.value)
-    local shakeX = math.random(-magnitude, magnitude)
-    local shakeY = math.random(-magnitude, magnitude)
-    self:moveTo(self.original_x + shakeX, self.original_y + shakeY)
+  for i, sprite in ipairs(buttonSprites) do
+    sprite:moveTo(spritePositions[i]:unpack())
+    sprite:setZIndex(100)
+    sprite:add()
+    sprite:setIgnoresDrawOffset(true)
   end
 
-  shakeTimer.timerEndedCallback = function()
-    self:moveTo(self.original_x, self.original_y)
-  end
+  self:setUpdatesEnabled(false)
 end
 
-function AbilityPanel:addItem(item)
-  if self.abilitiesCount == 3 then
-    table.remove(items, 1)
-    table.insert(items, item)
+-- gfx.sprite overrides to manage child button sprites along with sprite.
 
-    self:updateItemImages()
-  else
-    table.insert(items, item)
+function AbilityPanel:add()
+  AbilityPanel.super.add(self)
 
-    self:updateItemsCount()
-    self:updateItemImages()
+  for _, sprite in ipairs(buttonSprites) do
+    sprite:add()
   end
 end
 
-function AbilityPanel:cleanUp()
-  self:setItems()
-
-  self:remove()
-end
+function AbilityPanel:remove()
+  AbilityPanel.super.remove(self)
```

Comment:
should this be called after the loop that removes all the children? or does it not really matter

Reply:
Doesn't really matter... but we can do it for cleanliness :)